packages/@aws-cdk/aws-iam/test/merge-statements.test.ts

```diff
@@ -0,0 +1,328 @@
+import { App, Stack } from '@aws-cdk/core';
+import * as iam from '../lib';
+
+const PRINCIPAL_ARN = 'arn:aws:iam::111111111:user/user-name';
+const principal = new iam.ArnPrincipal(PRINCIPAL_ARN);
+
+const PRINCIPAL_ARN2 = 'arn:aws:iam::111111111:role/role-name';
+const principal2 = new iam.ArnPrincipal(PRINCIPAL_ARN2);
+
+test.each([
+  ['resources', true],
+  ['notResources', false],
+] as Array<['resources' | 'notResources', boolean]>)
+('merge %p statements: %p', (key, doMerge) => {
+  assertMergedP(doMerge, [
+    new iam.PolicyStatement({
+      [key]: ['a'],
+      actions: ['service:Action'],
+      principals: [principal],
+    }),
+    new iam.PolicyStatement({
+      [key]: ['b'],
+      actions: ['service:Action'],
+      principals: [principal],
+    }),
+  ], [
+    {
+      Effect: 'Allow',
+      Resource: ['a', 'b'],
+      Action: 'service:Action',
+      Principal: { AWS: PRINCIPAL_ARN },
+    },
+  ]);
+});
+
+test.each([
+  ['actions', true],
+  ['notActions', false],
+] as Array<['actions' | 'notActions', boolean]>)
+('merge %p statements: %p', (key, doMerge) => {
+  assertMergedP(doMerge, [
+    new iam.PolicyStatement({
+      resources: ['a'],
+      [key]: ['service:Action1'],
+      principals: [principal],
+    }),
+    new iam.PolicyStatement({
+      resources: ['a'],
+      [key]: ['service:Action2'],
+      principals: [principal],
+    }),
+  ], [
+    {
+      Effect: 'Allow',
+      Resource: 'a',
+      Action: ['service:Action1', 'service:Action2'],
+      Principal: { AWS: PRINCIPAL_ARN },
+    },
+  ]);
+});
+
+test.each([
+  ['principals', true],
+  ['notPrincipals', false],
+] as Array<['principals' | 'notPrincipals', boolean]>)
+('merge %p statements: %p', (key, doMerge) => {
```

Comment:
Can we get a more detailed descriptions of these tests? For example, some of the use identical Actions (`['service:Action']`), but some use different ones (`['service:Action1']` and `['service:Action2']`).

Would be great if the descriptions explained which test is which.